features/docs/stap-definities.feature

```diff
@@ -0,0 +1,326 @@
+#language: nl
+
+Functionaliteit: Stap definities
+
+  De titel van rule komt overeen met de naam van een ondersteunde 'gegeven' stap definitie
+  De scenario's beschrijven de SQL statements die worden gegenereerd voor de 'gegeven' stap definitie
+  In de scenario's is de pl_id nog niet opgenomen in de SQL statements
+
+  Rule: Gegeven de persoon met burgerservicenummer '<bsn>' heeft de volgende gegevens
+
+    Scenario: Persoon
+      Gegeven de persoon met burgerservicenummer '000000012' heeft de volgende gegevens
+      | naam                  | waarde |
+      | geslachtsnaam (02.40) | Jansen |
+      Dan zijn de gegenereerde SQL statements
+      | key          | text                                                                                                                                     | values                 |
+      | inschrijving | INSERT INTO public.lo3_pl(pl_id,mutatie_dt,geheim_ind) VALUES((SELECT MAX(pl_id)+1 FROM public.lo3_pl),current_timestamp,$1) RETURNING * | 0                      |
+      | persoon      | INSERT INTO public.lo3_pl_persoon(stapel_nr,volg_nr,persoon_type,burger_service_nr,geslachts_naam) VALUES($1,$2,$3,$4,$5)                | 0,0,P,000000012,Jansen |
```

Comment:
Hier moet je ook pl_id opnemen bij het toevoegen van een lo3_pl_persoon. Zit hieronder nog een magische toevoeging die niet in de feature staat?

Ik krijg hierop: ERROR:  null value in column "pl_id" of relation "lo3_pl_persoon" violates not-null constraint

Reply:
Zie opmerking in de feature description. Om deze feature te draaien moet je in world.js useDb op false zetten. De sql code wordt gegenereerd en niet uitgevoerd. En dat is de reden waarom er nog geen pl_id is toegevoegd. Die is er niet

Reply:
> Ik krijg hierop: ERROR: null value in column "pl_id" of relation "lo3_pl_persoon" violates not-null constraint

Ik neem aan dat je deze melding niet heb gekregen door het uitvoeren van de scenario's, maar dat je de statement hebt ge-copy/paste naar bijv. pgAdmin4 om de query te testen

Reply:
> Ik neem aan dat je deze melding niet heb gekregen door het uitvoeren van de scenario's, maar dat je de statement hebt ge-copy/paste naar bijv. pgAdmin4 om de query te testen

inderdaad copy/paste in pgAdmin. Maar moet het insert statement dat je in de feature beschrijft dan niet alsnog een pl_id bevatten? Wat getoond wordt in de feature is nu geen uitvoerbare sql opdracht

Reply:
Ik heb er voor gekozen om dat niet te doen, omdat je dan een pl_id moet hebben. Om dat netjes te doen moet ik dan ook een als stap toevoegen en nu sla ik dat over. Het ging er bij mij om of ik snap wat jullie willen. Maar goed, ik zal het toevoegen. Het betekent wel dat ik de aanschrijfwijze weer verder opschuift

Reply:
Als dat een bewuste keuze was en het verder werkt, vind ik dat ook prima, voor mij hoef je dat dan niet mooier te maken. Een korte toelichting dat ook pl_id wordt toegevoegd is dan ook genoeg